File: official/docs/java/current/batches/remove-shipments.java
package batches;

import com.easypost.exception.EasyPostException;
import com.easypost.model.Batch;
import com.easypost.service.EasyPostClient;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class RemoveShipments {
    public static void main(String[] args) throws EasyPostException {
        EasyPostClient client = new EasyPostClient(System.getenv("EASYPOST_API_KEY"));

        HashMap<String, Object> shipment1 = new HashMap<String, Object>();
        shipment1.put("id", "shp_...");

        List<HashMap<String, Object>> shipments = new ArrayList<HashMap<String, Object>>();
        shipments.add(shipment1);

        HashMap<String, Object> batchMap = new HashMap<String, Object>();
        batchMap.put("shipments", shipments);

        Batch batch = client.batch.removeShipments("batch_...", batchMap);

        System.out.println(batch);
    }
}

File: official/docs/golang/v4/batches/remove-shipments.go
package example

import (
	"fmt"

	"github.com/EasyPost/easypost-go/v4"
)

func removeShipments() {
	client := easypost.New("EASYPOST_API_KEY")

	shipment, _ := client.GetShipment("shp_...")

	batch, _ := client.RemoveShipmentsFromBatch("batch_...", shipment)

	fmt.Println(batch)
}
